thrift_rpc/recorder.go

```diff
@@ -385,6 +387,19 @@ func (r *Recorder) Flush() {
 	}
 }
 
+func (r *Recorder) Close() error {
+	r.lock.Lock()
+	closech := r.closech
+	r.closech = nil
+	r.lock.Unlock()
+
+	if closech != nil {
+		close(closech)
+	}
+
+	return nil
```

Comment:
Perhaps the right thing would be to first call `FlushLightStepTracer` before closing, but that API has similar problems (#89).